ensdomains/ens-app-v3
# update set primary name flow [reset]

**LATEST CHANGES V2**
* Refactored all the calculations for set primary name transactions into a hook called useGetPrimaryNameTransactionFlowItem. Hook uses a new pattern since it doesn't return data but a callback. This is needed to handle encrypted names.
* Added new hook useReverseRegistryName which will return the current set name for an account. This is needed to save gas if the user sets a primary name and then deletes or changes the eth record of that name.
* Removed a bunch of disabler conditions within hooks. I felt I added complexity and defeated the purpose of using react query.
* Removed a fast return results in useResolverStatus for when the resolver address is falsy or an empty address. Replaced it with fast return of auth data for resolvers that are known.

**LATEST CHANGES**

* Replaced "Migrate profile with eth record" to "Update eth record on latest resolver" since you can't migrate a profile if you are not authorized for it anyway.
* Added an option on useResolverStatus where you can see if a record is set on the latest resolver
* Will skip "update eth record on latest resolver" step if the latest resolver has eth record set.
* usePrimary returns null on failing check


---------------------------------
01_settings.spec.js
* remove wallet disconnect section

02_setPrimary.spec.js
* set primary name from settings page

05_permissions.spec, 06_extendNames.spec
* added waits and guards for CI stability

13_setPrimary.spec.js
* set primary name from profile

NameDetailsItem.tsx + test.tsx
* Update for safeDateObj

NameTableHeader
* Add styles for being in modal in select mode

SearchInput.tsx
* use the right query key

usePrimary, useNamesFromAddress
* refactor to be consistent with useQuery

useProfileActions
* Add additional checks for when to show the set primary name button as well as added support for decrypting names

MyNames, address
* refactor tables for updated useNamesFromAddress

Primary Section
* updated for latest UI

UnknownLabels-flow.tsx
* Break out the form into it's own component so it can be reused easily

Transaction Section
* Add modal to confirm clearing transactions

useAvailableNamesForAddress, useRegistryResolver
* New hooks needed for getting available primary names

array.ts
* moved chunkArr function here so it can be reused. Adjusted function so that it returns an empty nest array so that it mimics the return data of useInfiniteQuery.
